provider/provider_config/dynamo_config.go

```diff
@@ -0,0 +1,42 @@
+package provider_config
+
+import (
+	"encoding/json"
+
+	ss "github.com/featureform/helpers/string_set"
+)
+
+type DynamodbConfig struct {
+	Prefix    string
+	Region    string
+	AccessKey string
+	SecretKey string
+}
+
+func (d DynamodbConfig) Serialized() SerializedConfig {
+	config, err := json.Marshal(d)
+	if err != nil {
+		panic(err)
+	}
+	return config
+}
+
+func (d *DynamodbConfig) Deserialize(config SerializedConfig) error {
+	err := json.Unmarshal(config, d)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d DynamodbConfig) MutableFields() ss.StringSet {
+	return ss.StringSet{
+		"Region":    true,
```

Comment:
@simba-git and/or @sdreyer, now that I'm looking at this again, I'm not sure it makes sense to allow `Region` to change. My understanding is that you would have to actually replicate data to another instance in the new region, so this would likely be a new provider from our perspective, right?